app/analytics/query.ts

```diff
@@ -341,15 +356,18 @@ export class AnalyticsEngineAPI {
         page: number = 1,
         limit: number = 10,
     ) {
-        const intervalSql = intervalToSql(interval, tz);
+        const { startIntervalSql, endIntervalSql } = intervalToSql(
+            interval,
+            tz,
+        );
 
         const filterStr = filtersToSql(filters);
 
         const _column = ColumnMappings[column];
         const query = `
             SELECT ${_column}, SUM(_sample_interval) as count
             FROM metricsDataset
-            WHERE timestamp > ${intervalSql}
+            WHERE timestamp < ${endIntervalSql} AND timestamp >= ${startIntervalSql}
```

Comment:
Nit: I notice in some inconsistency in these `WHERE` clauses where sometimes you start the statement off with `timestamp >= ${startIntervalSql}` and sometimes you start them off with `timestamp < ${endIntervalSql}` (for example L231).

I'd prefer if it was consistent all the way through, starting with the `startInterval` boundary, and ending with the `endInterval` boundary.

Reply:
Nice catch! I have switched them around